File: tests/python/test_runner.py
#!/usr/bin/env python3

"""
Main test runner for Python tests
Runs all Python tests to verify documentation examples and commands
"""

import subprocess
import sys
import os
from pathlib import Path


def run_test_file(test_file):
    """Run a specific test file and return success status"""
    print(f"\n🏃 Running {test_file}...")
    
    try:
        result = subprocess.run([
            sys.executable, test_file
        ], capture_output=True, text=True, cwd=Path(__file__).parent)
        
        if result.returncode == 0:
            print(f"✅ {test_file} passed")
            print(result.stdout)
            return True
        else:
            print(f"❌ {test_file} failed")
            print("STDOUT:", result.stdout)
            print("STDERR:", result.stderr)
            return False
            
    except Exception as e:
        print(f"❌ Error running {test_file}: {e}")
        return False


def check_dependencies():
    """Check if required dependencies are available"""
    print("🔍 Checking dependencies...")
    
    dependencies = ['vaultsfyi', 'pytest']
    missing = []
    
    for dep in dependencies:
        try:
            __import__(dep)
            print(f"✅ {dep} is available")
        except ImportError:
            print(f"❌ {dep} is missing")
            missing.append(dep)
    
    return missing


def main():
    """Main test runner"""
    print("🚀 Running Python Tests for vaults.fyi Documentation\n")
    
    # Check if we're in a virtual environment
    in_venv = (
        hasattr(sys, 'real_prefix') or 
        (hasattr(sys, 'base_prefix') and sys.base_prefix != sys.prefix)
    )
    
    if in_venv:
        print("✅ Running in a virtual environment (recommended)")
    else:
        print("⚠️  Not running in a virtual environment")
        print("   Recommendation: Create and activate a virtual environment:")
        print("   python -m venv venv")
        print("   source venv/bin/activate  # On Windows: venv\\Scripts\\activate")
        print()
    
    # Check dependencies
    missing_deps = check_dependencies()
    if missing_deps:
        print(f"\n❌ Missing dependencies: {', '.join(missing_deps)}")
        print("Please install them with:")
        print("pip install " + " ".join(missing_deps))
        print("Or: pip install -r requirements.txt")
        return False
    
    print()
    
    # List of test files to run
    tests = [
        ('Virtual Environment Setup', 'test_venv_setup.py'),
        ('SDK Imports and Examples', 'test_imports.py'),
        ('Parameter Coverage Tests', 'test_parameters.py')
    ]
    
    print("📋 Test Plan:")
    for i, (name, file) in enumerate(tests, 1):
        print(f"   {i}. {name} ({file})")
    print()
    
    # Run all tests
    passed_tests = 0
    total_tests = len(tests)
    
    for test_name, test_file in tests:
        if run_test_file(test_file):
            passed_tests += 1
    
    # Summary
    print("\n" + "="*60)
    print(f"📊 Final Results: {passed_tests}/{total_tests} test suites passed")
    
    if passed_tests == total_tests:
        print("🎉 All Python tests passed!")
        print("✅ All documentation examples and commands are working correctly.")
        return True
    else:
        print("💥 Some tests failed.")
        print("❌ Documentation examples or commands may need to be updated.")
        return False


if __name__ == '__main__':
    success = main()
    sys.exit(0 if success else 1)

File: tests/js/test-runner.js
#!/usr/bin/env node

/**
 * Main test runner for JavaScript tests
 * Runs all JavaScript tests to verify documentation examples
 */

import { spawn } from 'child_process';
import { fileURLToPath } from 'url';
import { dirname, join } from 'path';

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

console.log('🚀 Running JavaScript Tests for vaults.fyi Documentation\n');

const tests = [
    { name: 'Import Tests', file: 'test-imports.js' },
    { name: 'Example Tests', file: 'test-examples.js' },
    { name: 'Parameter Coverage Tests', file: 'test-parameters.js' }
];

let passedTests = 0;
let totalTests = tests.length;

function runTest(testFile) {
    return new Promise((resolve) => {
        console.log(`\n🏃 Running ${testFile}...`);
        
        const testProcess = spawn('node', [join(__dirname, testFile)], {
            stdio: 'inherit'
        });
        
        testProcess.on('close', (code) => {
            if (code === 0) {
                passedTests++;
                console.log(`✅ ${testFile} passed\n`);
            } else {
                console.log(`❌ ${testFile} failed\n`);
            }
            resolve(code);
        });
        
        testProcess.on('error', (error) => {
            console.log(`❌ Error running ${testFile}: ${error.message}\n`);
            resolve(1);
        });
    });
}

async function runAllTests() {
    console.log('📋 Test Plan:');
    tests.forEach((test, index) => {
        console.log(`   ${index + 1}. ${test.name} (${test.file})`);
    });
    console.log('');
    
    for (const test of tests) {
        await runTest(test.file);
    }
    
    console.log('='.repeat(60));
    console.log(`📊 Final Results: ${passedTests}/${totalTests} test suites passed`);
    
    if (passedTests === totalTests) {
        console.log('🎉 All JavaScript tests passed!');
        console.log('✅ All documentation examples are working correctly.');
        process.exit(0);
    } else {
        console.log('💥 Some tests failed.');
        console.log('❌ Documentation examples may need to be updated.');
        process.exit(1);
    }
}

// Check if @vaultsfyi/sdk is installed
try {
    await import('@vaultsfyi/sdk');
    console.log('✅ @vaultsfyi/sdk is available');
} catch (error) {
    console.log('❌ @vaultsfyi/sdk is not installed.');
    console.log('   Please run: npm install @vaultsfyi/sdk');
    console.log('   Or run: npm install (to install from package.json)');
    process.exit(1);
}

runAllTests();